### DISCUSSION TITLE
Flask-SqlAlchemy + Websockets

### DISCUSSION THREAD

**Discussion by Exerosis** [pallets/quart#324] [General]
Right now I'm using flask + flask-sock + flask-sqllachemy and an issue I have is that each time a socket connects it takes a session from the pool and won't release that session until the socket disconnects. If the pool size is 10 then after 10 clients have connected the rest will timeout and crash.
I'm thinking of switching to quart for performance reasons, and I'm wondering if I will experience the same issue here or not.

**Comment by davidism** [comment]
You can return the connection to the pool at any time by calling `session.remove()`. Another connection will be claimed when needed. You'll probably have to do this regardless of the framework you use, since most framework db integrations are scoped to the life of the request, which isn't applicable with long running sockets.
